session-11/task/sync_mutex/task6.go

```diff
@@ -0,0 +1,30 @@
+package sync_mutex
+
+import (
+	"fmt"
+	"sync"
+)
+
+var studentGrades = make(map[string]int)
+
+func updateGrades(name string, score int, mux *sync.Mutex, wg *sync.WaitGroup) {
+
+	wg.Add(1)
+	mux.Lock()
+	studentGrades[name] = score
+	mux.Unlock()
+	defer wg.Done()
+
+}
+func Task6() {
+	fmt.Println("Task-6   **********************")
+	mux := sync.Mutex{}
+	wg := sync.WaitGroup{}
+
```

Comment:
Methodu burda cagirmagimiz dogrudur
```suggestion
   wg.add(4)
```